app/controllers/logins_controller.rb

```diff
@@ -4,4 +4,32 @@
 # a reasonable way to think about a "login" as a resource which is created
 # and deleted (i.e. 'logging out').
 # 
-# Reading and Updating a login, however, make a little less sense.
\ No newline at end of file
+# Reading and Updating a login, however, make a little less sense.
+
+MyApp.get "/" do
+  erb :"/homepage/homepage"
+end
+
+MyApp.get "/login" do
+  erb :"/logins/login"
+end
+
+MyApp.get "/login_attempt" do
+  @user = User.find_by_email(params[:email])
+  @user.name = params["name"]
```

Comment:
Why is `@user.name = params["name"]` here? What if there's no user with the given email address? That would cause a problem. Why is there even a `name` here to begin with?